packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -1969,6 +1970,42 @@ function renderRootSync(root: FiberRoot, lanes: Lanes) {
             workInProgressRootExitStatus = RootDidNotComplete;
             break outer;
           }
+          case SuspendedOnImmediate:
+          case SuspendedOnData: {
+            if (!didSuspendInShell) {
+              const handler = getSuspenseHandler();
+              if (handler === null) {
+                didSuspendInShell = true;
+                const counter = ++root.shellSuspendCounter;
+                if (counter > 100) {
+                  // This root has suspended repeatedly in the shell without
+                  // making any progress (i.e. committing something). This is
+                  // highly suggestive of an infinite ping loop, often caused by
+                  // an accidental Async Client Component.
+                  //
+                  // During a transition, we can suspend the work loop until the
+                  // promise to resolve, but this is a sync render, so that's
+                  // not an option. We also can't show a fallback, because none
+                  // was provided. So our last resort is to throw an error.
+                  //
+                  // TODO: Remove this error in a future release. Other ways
+                  // of handling this case include forcing a concurrent render,
+                  // or putting the whole root into offscreen mode.
+                  const asyncClientComponentError = new Error(
+                    'async/await is not yet supported in Client Components, ' +
+                      'only Server Components. This error is often caused by ' +
+                      "accidentally adding `'use client'` to a module that " +
+                      'was originally written for the server.',
```

Comment:
```suggestion
                    'Detected an infinite suspense loop while, ' +
                      'rendering the app shell. This error is often caused by ' +
                      "accidentally adding `'use client'` to a module that " +
                      'was originally written for the server.',
```

Reply:
I wanted the message to focus on the mistake that led to the crash. The fact that falls into a loop is an implementation detail; the reason for the error is we just don't support this feature yet.

Is the reason for the suggestion because technically there are other ways the error might occur?